umbraco-cms/tutorials/creating-an-xml-site-map.md

```diff
@@ -208,146 +232,169 @@ If we create a helper called `RenderSiteMapUrlEntriesForChildren` that accepts a
+8. Navigate to the **Content** section in the Umbraco backoffice.
+9. Open the Sitemap page and set the **Max Site Map Depth** to `2`.
+10. Save and publish the content.
 
-**Excluded DocumentType list**
+Your sitemap will now only contain entries for the top two levels. Leaving the value blank will mean that no maximum depth restriction will be applied.
 
-Our Xml Sitemap includes an entry for itself on the XML Sitemap, I thought we had excluded that document type, when we created the excludedDocumentType property...
+Finally, we need the helper to check the **Excluded Document Types** list on the XmlSiteMap Document Type.
 
-... we did but we haven't checked this value in the helper yet.
+1. Open the XmlSiteMap Template.
+2. Add the following code snippets within the first set of curly brackets:
 
-```csharp
-string excludedDocumentTypeList = Model.Value<string>("excludedDocumentTypes");
-string[] excludedDocumentTypes = (!String.IsNullOrEmpty(excludedDocumentTypeList)) ? excludedDocumentTypeList.Split(new char[] { ',' }, StringSplitOptions.RemoveEmptyEntries).Select(p => p.Trim()).ToArray() : new string[] { };
-```
+    ```csharp
+    // Get the value from the excludedDocumentTypes property as a 'string'
+    string excludedDocumentTypeList = Model.Value<string>("excludedDocumentTypes");
+    // Separate the values into separate Document Types and add them to an 'array'
+    string[] excludedDocumentTypes = (!String.IsNullOrEmpty(excludedDocumentTypeList)) ? excludedDocumentTypeList.Split(new char[] { ',' }, StringSplitOptions.RemoveEmptyEntries).Select(p => p.Trim()).ToArray() : new string[] { };
+    ```
 
-now we can pass this value into our helper
+3. Update the `RenderSiteMapUrlEntriesForChildren` helper as shown below to pass in the array:
 
-```csharp
-void RenderSiteMapUrlEntriesForChildren(IPublishedContent parentPage)
-{
-    foreach (var page in parentPage.Children.Where(f => !excludedDocumentTypes.Contains(f.ContentType.Alias) && !f.Value<bool>("hideFromXmlSiteMap")))
+    ```csharp
+    void RenderSiteMapUrlEntriesForChildren(IPublishedContent parentPage)
     {
-        @RenderSiteMapUrlEntry(page)
-        if (page.Level < maxSiteMapDepth && page.Children.Any(f => !excludedDocumentTypes.Contains(f.ContentType.Alias) && !f.Value<bool>("hideFromXmlSiteMap")))
+        // Filter the query based on the excludedDocumentTypes value
+        foreach (var page in parentPage.Children.Where(f => !excludedDocumentTypes.Contains(f.ContentType.Alias) && !f.Value<bool>("hideFromXmlSiteMap")))
         {
-            @RenderSiteMapUrlEntriesForChildren(page, maxSiteMapDepth, excludedDocumentTypes)
+            RenderSiteMapUrlEntry(page);
+            if (page.Level < maxSiteMapDepth && page.Children.Any(f => !f.Value<bool>("hideFromXmlSiteMap")))
+            {
+                RenderSiteMapUrlEntriesForChildren(page, maxSiteMapDepth, excludedDocumentTypes);
```

Comment:
This line throws an error:

![image](https://user-images.githubusercontent.com/82437098/211560753-ab2d038a-a8ca-4623-9831-85a8427b6f1d.png)

Reply:
Yes, the parameters need to be taken out - I'll update the sample 💪